### ISSUE TITLE
No way to get status_code or error

### ISSUE THREAD

**Issue by xssc** [nabeelamjad/poke-api#14]
There is no way to get the `status_code` of a response or the `error` string because the response given only includes the subresponses

**Comment by xssc** [comment]
Replacing [These lines](https://github.com/nabeelamjad/poke-api/blob/master/lib/poke-api/response.rb#L27-L28) with something like the code below helps, but I was running into some issues when I tested it and am busy right now

```
@resonse = @response.to_hash.delete('returns')
@response = decoded_resp.merge(@response)
```

**Comment by nabeelamjad** [comment]
I'll add the `status_code` and `error` in, going to leave out the authentication information unless there's demand for it (it's in bytes anyhow).

**Comment by xssc** [comment]
Okay, thanks

**Comment by nabeelamjad** [comment]
Commit 8f3caa91adb07c1e636dd9e9b71cfec4ae0bde5b fixes this issue, new tag is 0.0.4 with updated readme/gpsoauth and response.

You can access the status code and error using `call.response.status_code` and `call.response.error` (if the error is empty then assume there's no error).
